bundles/org.openhab.core/src/test/java/org/openhab/core/library/unit/CurrencyUnitTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.library.unit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.junit.jupiter.api.Test;
+import org.openhab.core.internal.library.unit.CurrencyConverter;
+import org.openhab.core.internal.library.unit.CurrencyService;
+import org.openhab.core.library.dimension.Currency;
+import org.openhab.core.library.types.QuantityType;
+
+/**
+ * The {@link CurrencyConverter} is a
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class CurrencyUnitTest {
+
+    @SuppressWarnings("unused")
+    private final CurrencyService currencyService = new CurrencyService(new TestCurrencyProvider());
+
+    @Test
+    public void testSimpleConversion() {
+        QuantityType<Currency> amount = new QuantityType<>("7.45 DKK");
+        QuantityType<Currency> convertedAmount = amount.toUnit("€");
+
+        assertThat(convertedAmount, is(notNullValue()));
+        assertThat(convertedAmount.getUnit(), is(TestCurrencyProvider.EUR));
+        assertThat(convertedAmount.doubleValue(), closeTo(1, 1e-4));
+    }
+
+    @Test
+    public void testInverseConversion() {
+        QuantityType<Currency> amount = new QuantityType<>("1.00 €");
+        QuantityType<Currency> convertedAmount = amount.toUnit("DKK");
+
+        assertThat(convertedAmount, is(notNullValue()));
+        assertThat(convertedAmount.getUnit(), is(TestCurrencyProvider.DKK));
+        assertThat(convertedAmount.doubleValue(), closeTo(7.45, 1e-4));
+    }
+
+    @Test
+    public void testComplexConversion() {
+        QuantityType<Currency> amount = new QuantityType<>("13 DKK");
+        QuantityType<Currency> convertedAmount = amount.toUnit("$");
+
+        assertThat(convertedAmount, is(notNullValue()));
+        assertThat(convertedAmount.getUnit(), is(TestCurrencyProvider.USD));
+        assertThat(convertedAmount.doubleValue(), closeTo(1.8845, 1e-4));
+    }
+
+    @Test
+    public void testPriceCalculation() {
+        QuantityType<?> unitPrice = new QuantityType<>("0.25 €/kWh");
```

Comment:
Shouldn't this only work for "0.25 EUR/kWh", since the symbol is ambiguous and we're using the currency code as key?

Reply:
The `CurrencyProvider` has to take care that the symbol is unique. For the system provider there is only one currency, so there can't be ambiguity. For (future) extensions, the provider has to make sure that e.g. `Kr` or `$` is only used for one currency and either not set a symbol for other currencies or prefix them (like `AUS$`).

Reply:
Hmm, it's quite possible (even likely) I don't fully understand it yet. Let's say you have two currencies:
- Currency code: NOK, symbol: kr
- Currency code: SEK, symbol: kr

I then receive an energy price in NOK, e.g. 1 NOK/kWh, and would like that converted (through an exchange rate service) to SEK. How would I then define that quantity, if not like `new QuantityType<>("1 NOK/kWh")`? I imagined that symbols would only be used for something like state patterns, i.e. only for display purposes.

Reply:
I didn't mean to test that, only to ensure coverage for using a currency code rather than currency symbol. So simply the same testcase but using for example "EUR" instead of "€".